QualityControl/public/common/filters/filter.js

```diff
@@ -0,0 +1,134 @@
+ * @param {string} [config.type='text'] - The type of the filter element (e.g., 'text', 'number').
+ * @param {RemoteData} [config.options=RemoteData.notAsked()] - List of options for a dropdown selector (optional).
+ * @param {Function} config.onChangeCallback - Callback to be triggered on the change event of the filter.
+ * @param {Function} config.onInputCallback - Callback to be triggered on the input event.
+ * @param {Function} config.onEnterCallback - Callback to be triggered when the Enter key is pressed.
+ * @param {string} [config.filterType=FILTER_TYPE.DROPDOWN] - The type of filter to be used.
+ * @param {string} [config.width='w-20'] - The CSS class that defines the width of the filter.
+ * @returns {vnode} - A virtual node element representing the filter element (input or dropdown).
+ */
+const dynamicSelector = (config) => {
+  const {
+    queryLabel,
+    placeholder,
+    id, filterMap,
+    options = RemoteData.notAsked(),
+    onChangeCallback,
+    onInputCallback,
+    onEnterCallback,
+    filterType = FILTER_TYPE.DROPDOWN,
+    type = 'text',
+    width = 'w-20',
+  } = config;
+
+  const renderFilterInput = () =>
+    filterInput({ queryLabel, placeholder, id, filterMap, onInputCallback, onEnterCallback, type, width });
+
+  return options.match({
+    Success: (optionsList) => {
+      if (filterType === FILTER_TYPE.DROPDOWN && optionsList.length > 0) {
+        return dropdownSelector({
+          queryLabel, placeholder, id, filterMap, options: optionsList, onChangeCallback, width,
+        });
+      }
+      return renderFilterInput();
+    },
+    Other: renderFilterInput,
+  });
+};
+
+/**
+ * Builds a filter input element that allows the user to specify a parameter to be used when querying objects.
+ * This function renders a text input element with event handling for input and Enter key press.
+ * @param {object} config - Configuration object for building the filter input element.
+ * @param {string} config.queryLabel - The key used to query the storage with this parameter.
+ * @param {string} config.placeholder - The placeholder text to be displayed in the input field.
+ * @param {string} config.id - The unique identifier for the input field.
+ * @param {object} config.filterMap - Map of the current filter values.
+ * @param {Function} config.onInputCallback - Callback to be triggered on the input event.
+ * @param {Function} config.onEnterCallback - Callback to be triggered when the Enter key is pressed.
+ * @param {string} [config.type='text'] - The type of the filter element (e.g., 'text', 'number').
+ * @param {string} [config.width='w-20'] - The CSS class that defines the width of the filter.
+ * @returns {vnode} - A virtual node element representing the filter input.
+ */
+const filterInput = (config) => {
+  const { queryLabel, placeholder, id, filterMap, onInputCallback, onEnterCallback, type, width = 'w-20' } = config;
+
+  return h(`${width}`, [
```

Comment:
You are missing the `.` in front of the class. 
Like you wrote it, it will create an HTML element w-20 
```html
<w-20></w-20>
```
(the same goes for the dropdown selector